conda/shell/etc/profile.d/conda.sh

```diff
@@ -42,6 +42,11 @@ conda() {
             activate|deactivate)
                 __conda_activate "$cmd" "$@"
                 ;;
+            exec)
+                \local env=$1
+                shift
+                _conda_activate "$env" && exec $@
```

Comment:
That last comment implies it's ultimately easier to do this *right*.  We actually don't need #6820 to complete this, even though it would be nice. The right way to do this though is to keep the majority of the code in python land, use something like https://github.com/tox-dev/tox-conda/blob/40fbb1869fca973b656e461147a48fa7af245e6d/tox_conda/plugin.py#L55-L83 to get the activated environment variables for the process, set them in the current process, then use one of the python [`os.exec`](https://docs.python.org/2/library/os.html#os.execl) functions.

Reply:
Oh, and we'll also need to deal with `conda exec --help`.  Otherwise, how are people going to know the the two positional arguments should be, and what order they should be in?

Reply:
@kale I'm not convinced I need to fork anything. I've actually tested this and it seems to work just fine.